sdcflows/transform.py

```diff
@@ -32,15 +59,166 @@
 
 import nibabel as nb
 import nitransforms as nt
-from nitransforms.base import _as_homogeneous
 from bids.utils import listify
 
 from niworkflows.interfaces.nibabel import reorient_image
 
+from sdcflows.utils.tools import ensure_positive_cosines
+
+
+def _sdc_unwarp(
+    data: np.ndarray,
+    coordinates: np.ndarray,
+    pe_info: Tuple[int, float],
+    hmc_xfm: np.ndarray,
```

Comment:
since the in/ex-clusion of this is important
```suggestion
    hmc_xfm: Optional[np.ndarray],
```